### ISSUE TITLE
Cannot get it working with Pi 2 hardware

### ISSUE THREAD

**Issue by lukebutcher** [orgicus/ofxCvPiCam#1]
Hi, I've used this excellent library on a Pi 1, however I cannot get it working at all on the Pi 2. I cannot even get the examples to work. The ofxRPiCameraVideoGrabber library works fine on the Pi 2, so I don't think it's a camera issue. Are you able to help me, please? Thank you.

**Comment by orgicus** [comment]
Hi,

I've actually recently got my PI 2 and it works.
I did encounter some issues, these were with the OpenCV libraries themselves, 
not the camera or the driver code I'm using (which is why ofxRPiCameraVideoGrabber works).
(For example if you try to compile the ofxOpenCV or ofxCv examples alone you'll run into the same problems, regardless ofxCvPiCam)

You can see some details on this issue here:
http://forum.openframeworks.cc/t/unable-to-link-opencvexample-on-armv7/14786
Long story short, until the OF's OpenCV static libraries get updated, 
you can download the static libraries Aldo Hoeben recompiled from here:
https://github.com/fieldOfView/openFrameworks/tree/a1077f19ca7129cd7a5fbdea2dca1934875dcdc2/addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l
Replace the old ones from the addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l/

Let me know how this goes or if you run into any other issues.

Good luck,
George

**Comment by lukebutcher** [comment]
Hello George,

Excellent – thank you for such a fast response!

I will follow your advice – and confirm how I fare.

Thanks again – your advice is very much appreciated!

Luke

From: George Profenza
Reply-To: orgicus/ofxCvPiCam
Date: Thursday, 2 April 2015 16:18
To: orgicus/ofxCvPiCam
Cc: Luke Butcher
Subject: Re: [ofxCvPiCam] Cannot get it working with Pi 2 hardware (#1)

Hi,

I've actually recently got my PI 2 and it works.
I did encounter some issues, these were with the OpenCV libraries themselves,
not the camera or the driver code I'm using (which is why ofxRPiCameraVideoGrabber works).

You can see some details on this issue here:
http://forum.openframeworks.cc/t/unable-to-link-opencvexample-on-armv7/14786
Long story short, until the OF's OpenCV static libraries get updated,
you can download the static libraries Aldo Hoeben recompiled from here:
https://github.com/fieldOfView/openFrameworks/tree/a1077f19ca7129cd7a5fbdea2dca1934875dcdc2/addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l
Replace the old ones from the addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l/

Let me know how this goes or if you run into any other issues.

Good luck,
George

**Comment by lukebutcher** [comment]
Perfect – that’s made my day!

Thank you VERY much!

Best wishes,

Luke

From: George Profenza
Reply-To: orgicus/ofxCvPiCam
Date: Thursday, 2 April 2015 16:18
To: orgicus/ofxCvPiCam
Cc: Luke Butcher
Subject: Re: [ofxCvPiCam] Cannot get it working with Pi 2 hardware (#1)

Hi,

I've actually recently got my PI 2 and it works.
I did encounter some issues, these were with the OpenCV libraries themselves,
not the camera or the driver code I'm using (which is why ofxRPiCameraVideoGrabber works).

You can see some details on this issue here:
http://forum.openframeworks.cc/t/unable-to-link-opencvexample-on-armv7/14786
Long story short, until the OF's OpenCV static libraries get updated,
you can download the static libraries Aldo Hoeben recompiled from here:
https://github.com/fieldOfView/openFrameworks/tree/a1077f19ca7129cd7a5fbdea2dca1934875dcdc2/addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l
Replace the old ones from the addons/ofxOpenCv/libs/opencv/lib/linuxarmv7l/

Let me know how this goes or if you run into any other issues.

Good luck,
George

**Comment by orgicus** [comment]
Awesome!  Glad it works on your PI 2.

Hopefully the updated static libraries will make it to the main OF repo
and this step won't be necessary (otherwise a script might shave off a few minutes)

Thanks for the update, I'll close this issue now.
